src/mon/OSDMonitor.cc

```diff
@@ -75,6 +75,59 @@
 #define dout_subsys ceph_subsys_mon
 #define OSD_PG_CREATING_PREFIX "osd_pg_creating"
 
+void LastEpochClean::Lec::report(ps_t ps, epoch_t last_epoch_clean)
+{
+  if (epoch_by_pg.size() <= ps) {
+    epoch_by_pg.resize(ps + 1, 0);
+  }
+  if (epoch_by_pg[ps] >= last_epoch_clean) {
+    // stale lec
+    return;
+  }
+  epoch_by_pg[ps] = last_epoch_clean;
+  if (last_epoch_clean < floor) {
+    floor = last_epoch_clean;
```

Comment:
no, we are conservative here. even if there is one or more pgs whose lec is between floor and ps. we use the smallest lec. so we don't trim the epochs still being used by the pg whose lec is the smallest.